apps/meeting/src/components/MeetingControls/VideoInputTransformControl.tsx

```diff
@@ -0,0 +1,238 @@
+  backgroundReplacementLabel?: string;
+}
+
+const VideoInputTransformControl: React.FC<Props> = ({
+  label = 'Video',
+  backgroundBlurLabel = 'Enable Background Blur',
+  backgroundReplacementLabel = 'Enable Background Replacement',
+}) => {
+  const meetingManager = useMeetingManager();
+  const { devices, selectedDevice } = useVideoInputs();
+  const { isVideoEnabled, toggleVideo } = useLocalVideo();
+  const { isBackgroundBlurSupported, createBackgroundBlurDevice } = useBackgroundBlur();
+  const { isBackgroundReplacementSupported, createBackgroundReplacementDevice } = useBackgroundReplacement();
+  const [isLoading, setIsLoading] = useState(false);
+  const [dropdownWithVideoTransformOptions, setDropdownWithVideoTransformOptions] = useState<ReactNode[] | null>(null);
+  const [activeVideoDevice, setDevice] = useState<Device | VideoTransformDevice | null>(meetingManager.selectedVideoInputTransformDevice as Device);
+  const [activeTransform, setActiveTransform] = useState<string>(VIDEO_TRANSFORM_OPTIONS.none);
+
+  const videoDevices: DeviceType[] = useMemoCompare(devices, (prev: DeviceType[] | undefined, next: DeviceType[] | undefined): boolean => isEqual(prev, next));
+
+  useEffect(() => {
+    meetingManager.subscribeToSelectedVideoInputTransformDevice(setDevice);
+    resetDeviceToIntrinsic();
+    return () => {
+      meetingManager.unsubscribeFromSelectedVideoInputTranformDevice(setDevice);
+    };
+  }, []);
+
+  // Reset the video input to intrinsic if current video input is a transform device because this component does not know if blur or replacement was selected.
+  // This depends on how the demo is set up.
+  // TODO: use a hook in the appState to track whether blur or replacement was selected before this component mounts.
+  async function resetDeviceToIntrinsic() {
+    let current = activeVideoDevice;
+    if (isVideoTransformDevice(current)) {
+      const intrinsicDevice = await current.intrinsicDevice();
+      current = intrinsicDevice;
+      await meetingManager.selectVideoInputDevice(current);
+    }
+  }
+
+  // Toggle background blur on/off.
+  async function toggleBackgroundBlur() {
+    let current = activeVideoDevice;
+    
+    if (isLoading) {
+      return;
+    }
+    setIsLoading(true);
+
+    if (!isVideoTransformDevice(current)) {
+      // Enable video transform on the default device.
+      current = await createBackgroundBlurDevice(current) as VideoTransformDevice;
+      meetingManager.logger?.info('Video filter turned on - selecting video transform device: ' + JSON.stringify(current));
+    } else {
+      // Switch back to intrinsicDevice.
+      const intrinsicDevice = await current.intrinsicDevice();
+      // Stop existing VideoTransformDevice.
+      await current.stop();
+
+      current = intrinsicDevice;
+      
+      // Switch to background blur device if old selection was background replacement otherwise switch to default intrinsic device.
```

Comment:
`toggleBackgroundBlur` and `toggleBackgroundReplacement` have an almost identical implementation, but they are not scalable. For example, if we add a new filter method, say `toggleBackgroundVideo`, then we need to have another cloned method with more complex logic.

Reply:
You can skip this comment in this PR. In the future, we would rewrite them like the following:
- One shared method that takes in a different value as a method parameter
- Shouldn't the `activeTransform` state be a map of <transform, selected>? So when we add a new filter, we can iterate this map and re-apply existing filters.

Reply:
I agree, I did also notice this problem. `PopOverItem` onClick only accepts `() => void` so I couldn't pass a parameter when I was trying to use on shared method but I just figured out a different solution to it. I will take this as a SIM to clean it up and make it scalable.